code/mob/living/critter/skeleton.dm

```diff
@@ -81,5 +106,76 @@
 		HH.icon_state = "handr"
 
 	setup_healths()
-		add_hh_flesh(50, 1)
-		add_hh_flesh_burn(50, 0.7)
+		add_hh_flesh(src.health_brute, src.health_brute_vuln)
+		add_hh_flesh_burn(src.health_burn, src.health_brute_vuln)
+
+	seek_target(var/range = 6)
+		. = list()
+		for (var/mob/living/C in hearers(range, src))
+			if (isdead(C)) continue
+			if (isintangible(C)) continue //don't attack what you can't touch
+			if (istype(C, /mob/living/critter/skeleton)) continue
+			if (iswizard(C) && src.wizardSpawn) continue
+			. += C
+
+	critter_attack(mob/target)
+		var/datum/targetable/critter/tackle = src.abilityHolder.getAbility(/datum/targetable/critter/tackle)
+		if (!tackle.disabled && tackle.cooldowncheck())
+			tackle.handleCast(target)
+		else
+			return ..()
+
+	death(var/gibbed)
+		if (prob(src.revivalChance))
+			..()
+			src.revivalChance -= src.revivalDecrement
+			SPAWN(rand(400,800))
+				src.full_heal()
+				src.set_density(1)
+				src.visible_message("<span class='alert'>[src] re-assembles and is ready to fight once more!</span>")
+			return
+		if (!gibbed)
+			src.visible_message("<span class='alert'>[src] explodes into bones!</span>")
+			src.unequip_all()
+			src.gib()
+		return ..()
+
+
+	proc/CustomiseSkeleton(var/mob/living/carbon/human/target, var/is_monkey)
+		src.name = "[capitalize(target)]'s skeleton"
+		src.desc = "A horrible skeleton, raised from the corpse of [target] by a wizard."
+		src.revivalChance = 100
+		src.wizardSpawn = TRUE
+
+		if (is_monkey)
+			icon = 'icons/mob/monkey.dmi'
+		return
+
+/mob/living/critter/skeleton/wraith
+	desc = "It looks rather crumbly."
+	icon = 'icons/mob/human_decomp.dmi'
+	icon_state = "decomp4"
+	health_brute = 15
+	health_burn = 15
+
+	seek_target(var/range = 7)
+		. = list()
+		for (var/mob/living/C in hearers(range, src))
+			if (isdead(C)) continue
+			if (isintangible(C)) continue //don't attack what you can't touch
+			if (islivingobject(C)) continue //don't attack wraith objects
+			if (istype(C, /mob/living/critter/wraith)) continue // don't attack wraith summons
+			if (istype(C, /mob/living/critter/skeleton)) continue
+			. += C
+
+	death()
+		..()
+		particleMaster.SpawnSystem(new /datum/particleSystem/localSmoke("#000000", 5, locate(x, y, z)))
```

Comment:
```suggestion
		particleMaster.SpawnSystem(new /datum/particleSystem/localSmoke("#000000", 5, get_turf(src)))
		..()
```
Weird to use `locate` here when you can just get the turf under the mob (I actually don't know which is faster but this one is more readable).
Also moved to above the parent call, since we gib/delete them in the parent, which might cause issues getting the turf (since we might be in `null` already).